docs/oauth.md

```diff
@@ -0,0 +1,169 @@
+# OAuth2 API
+
+OAuth2 is a protocol that allows applications to use Webmaker accounts without needing to store a username and password.
+
+Before you can use OAuth2 you'll need to register your application.
+
+## Registering your application
+
+Webmaker requires the following information to register your application:
+
+* **Name** - The friendly name of your application e.g. "Buckley's Bees".
+* **Description** - A description of your application e.g. "Buckley's Bees sells the best honey in Ontario".
+* **Image** - An image that represents your application.
+* **Homepage URL** - The full URL to your application homepage e.g. "https://buckleysbees.ca".
+* **Redirect URL** - Your application's OAuth2 callback URL e.g. "https://buckleysbees.ca/oauth2/callback".
+
+Once you've registered your application you'll recieve your application credentials:
+
+* **Client ID** - A string identifying your application. This value is not secret and can be shown publicly e.g. `wm_id_BTQNPABtUqqApaDrcsDa`.
+* **Client Secret** - A string. This value is secret and must not be shared with anyone e.g. `wm_secret_iHmAWhvCwBYnuE6aZHcArBEPohanKmmWr8LcyBBPZYRkAxdmui`.
+
+You can register your application by emailing [jbuck](mailto:jon@mozillafoundation.org). Self-serve application registration is coming shortly.
```

Comment:
okay so yes, that's the current way to do it, but if we have any idea how to do it later, let's already doc that =)